stackslib/src/net/http.rs

```diff
@@ -3978,48 +4043,50 @@ impl HttpResponseType {
     }
 
     pub fn metadata(&self) -> &HttpResponseMetadata {
```

Comment:
Can you undo the whole `*self` --> `self` transformation here and elsewhere?  It's gonna make the merge conflict that much worse when we merge this into the HTTP RPC refactor PR.  That goes for any beatification of this file -- it's going to disappear in `develop` so there's no use in trying to make it look better ;)